CMakeLists.txt

```diff
@@ -300,10 +300,14 @@ if(ENABLE_ASAN)
     if (NOT ENABLE_ASAN_OK)
         MESSAGE(FATAL_ERROR "The compiler does not support address sanitizer")
     endif()
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    add_compile_options(-static-libasan)
+    add_link_options(-static-libasan)
+endif()
```

Comment:
Is there any other workaround to deal with the problem you encountered?
As far as I know, statically linking against libasan is problematic for some GCC versions.

Reply:
Ok, i will added the options to control this.